src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -268,9 +268,9 @@ public void Clear()
                 _count = 0;
                 _freeList = -1;
                 _freeCount = 0;
-                _version++;
                 Array.Clear(_entries, 0, count);
             }
+            _version++;
```

Comment:
Why is this an improvement? I would expect that it prevents tailcall for `Array.Clear`. Should the version increment be done as the first thing, similar to insert?

Reply:
Doesn't reduce the size, though keeping the tail call have moved now the two Clears together at end so it can just run through cache setting them before making the calls out
```asm
; Lcl frame size = 40
G_M52189_IG01:
       push     rdi
       push     rsi
       sub      rsp, 40
       mov      rsi, rcx
G_M52189_IG02:
       mov      edi, dword ptr [rsi+56]
       test     edi, edi
       jle      SHORT G_M52189_IG04
       xor      r8d, r8d
       mov      dword ptr [rsi+56], r8d
       mov      dword ptr [rsi+60], -1
       mov      dword ptr [rsi+64], r8d
       inc      dword ptr [rsi+68]
       mov      rcx, gword ptr [rsi+8]
       mov      r8d, dword ptr [rcx+8]
       xor      edx, edx
       call     Array:Clear(ref,int,int)
       mov      rcx, gword ptr [rsi+16]
       mov      r8d, edi
       xor      edx, edx
       lea      rax, [(reloc)]
G_M52189_IG03:
       add      rsp, 40
       pop      rsi
       pop      rdi
       rex.jmp  rax
G_M52189_IG04:
       add      rsp, 40
       pop      rsi
       pop      rdi
       ret      
; Total bytes of code 84, prolog size 9 for method Dictionary`2:Clear():this
```